src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ReadOnlyTensorSpan.cs

```diff
@@ -166,7 +167,7 @@ public ReadOnlyTensorSpan(Array? array, scoped ReadOnlySpan<int> start, scoped R
                 this = default;
                 return; // returns default
             }
-            if (!typeof(T).IsValueType && array.GetType() != typeof(T[]))
+            if (array.GetType().GetElementType() != typeof(T))
```

Comment:
This looks like a correct fix. But I'd like us to get an issue logged around this, as this seems like a pit of failure here.

In particular, since `Array` is "untyped" and due to how `new[] {...}` works, it means that users can get into these types of silent failures; when they likely expected it to be implicitly typed as `double[]` like `[...]` would've been.

So I think we may want to consider if this should be a constructor or instead some named `Create*` method where `*` is an appropriate suffix.